### PR TITLE
Refactor Data and Messages

### PR DESCRIPTION
**Pull request by rjkiv**
-make the DataNode value union anonymous and removed the RawVal function
-added "Fast" methods for DataNode/Array to satisfy retail calls
-added brief descriptions for Data stuff for doxygen
-moved Message declarations from header files such as "UIMessages.h", "GameMessages.h", to headers relevant to those messages (so for example, UIComponentScrollMsg was moved to UIComponent.h)
-refactored the Message macro to avoid having to make inline ctor definitions outside of the class body

### CODE REVIEW

File: src/system/obj/Data.h [hunk 1]
```diff
@@ -119,10 +121,16 @@ class DataNode {
...
-    DataNodeValue RawVal() { return mValue; }
     bool CompatibleType(DataType) const;
     DataNode& Evaluate() const;
 
+    // these were implemented to match up in retail
+    // please do not use these in regular code
+    int FastInt() const { return mValue.integer; }
+    const char* FastStr() const { return mValue.symbol; }
+    Hmx::Object* FastObj() const { return mValue.object; }
+    DataArray* FastArray() const { return mValue.array; }
```
Comment (TheNathannator) [inline src/system/obj/Data.h:132]: I'd say these should be labelled as `Unchecked` rather than `Fast`, to better indicate the intended use case.

Comment (rjkiv) [reply src/system/obj/Data.h:132]: Good idea, I think I'll rename it to `Unsafe` to indicate that it's, well, unsafe, and you shouldn't use these ordinarily lol

Comment (rjkiv) [reply src/system/obj/Data.h]: pushed

Comment (TheNathannator) [reply src/system/obj/Data.h]: That works too :+1:
I've just typically seen `Unchecked` used for this kind of thing, to imply that you are responsible for doing the necessary checks up-front.

Comment (rjkiv) [reply src/system/obj/Data.h]: since Unchecked is what's typically used, I went ahead and renamed that to match common conventions

**Pull request merged**